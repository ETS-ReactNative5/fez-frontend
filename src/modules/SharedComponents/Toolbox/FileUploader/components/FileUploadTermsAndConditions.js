import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {withStyles} from '@material-ui/core/styles';
import {FormControlLabel, Typography, Checkbox} from '@material-ui/core';

export class FileUploadTermsAndConditions extends PureComponent {
    static propTypes = {
        isTermsAndConditionsAccepted: PropTypes.bool,
        onAcceptTermsAndConditions: PropTypes.func,
        classes: PropTypes.object,
        accessTermsAndConditions: PropTypes.string,
        disabled: PropTypes.bool
    };

    _handleChange = (event) => {
        this.props.onAcceptTermsAndConditions(event.target.checked);
    };

    render() {
        const {isTermsAndConditionsAccepted, classes, accessTermsAndConditions, disabled} = this.props;

        return (
            <FormControlLabel
                classes={{
                    root: classes.root
                }}
                disabled={disabled}
                control={
                    <Checkbox
                        checked={isTermsAndConditionsAccepted}
                        onChange={this._handleChange}
                        classes={{root: classes.checkboxRoot, checked: classes.checkboxChecked}}
                    />
                }
                label={
                    <Typography
                        classes={{
                            root: classes.label
                        }}
                        color={!isTermsAndConditionsAccepted ? 'error' : 'secondary'}
                    >
                        {accessTermsAndConditions}
                    </Typography>
                }
            />
        );
    }
}

const styles = (theme) => ({
    root: {
        alignItems: 'flex-start'
    },
    label: {
        textAlign: 'justify',
        fontSize: 16,
        fontWeight: 300,
        lineHeight: '24px',
        paddingTop: 10
    },
    checkboxRoot: {
        color: theme.status.danger,
    },
    checkboxChecked: {
        color: `${theme.palette.primary.main} !important`
    }
});

export default withStyles(styles)(FileUploadTermsAndConditions);
