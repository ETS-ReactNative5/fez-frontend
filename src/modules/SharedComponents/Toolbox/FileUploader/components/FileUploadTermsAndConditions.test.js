import { FileUploadTermsAndConditions } from './FileUploadTermsAndConditions';

function setup(testProps, isShallow = true) {
    const props = {
        disabled: false,
        isTermsAndConditionsAccepted: false,
        onAcceptTermsAndConditions: jest.fn(),
        acceptTermsAndConditions: 'test terms and conditions',
        classes: {
            label: '',
            error: '',
            accepted: ''
        },
        ...testProps
    };

    return getElement(FileUploadTermsAndConditions, props, isShallow);
}

describe('Component FileUploadTermsAndConditions', () => {
    it('should render default view', () => {
        const wrapper = setup({});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render checked if terms and conditions accepted', () => {
        const wrapper = setup({isTermsAndConditionsAccepted: true});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should call onAcceptTermsAndConditions to handle change', () => {
        const testFn = jest.fn();
        const wrapper = setup({onAcceptTermsAndConditions: testFn});
        wrapper.instance()._handleChange({target: {checked: true}});
        expect(testFn).toHaveBeenCalledWith(true);
    });
});