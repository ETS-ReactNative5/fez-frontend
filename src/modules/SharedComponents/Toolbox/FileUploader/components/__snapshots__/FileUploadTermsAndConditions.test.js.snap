// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component FileUploadTermsAndConditions should render checked if terms and conditions accepted 1`] = `
<WithStyles(FormControlLabel)
  classes={
    Object {
      "root": undefined,
    }
  }
  control={
    <WithStyles(Checkbox)
      checked={true}
      classes={
        Object {
          "checked": undefined,
          "root": undefined,
        }
      }
      onChange={[Function]}
    />
  }
  disabled={false}
  label={
    <WithStyles(Typography)
      classes={
        Object {
          "root": Object {
            "textAlign": "justify",
          },
        }
      }
      color="secondary"
    />
  }
/>
`;

exports[`Component FileUploadTermsAndConditions should render default view 1`] = `
<WithStyles(FormControlLabel)
  classes={
    Object {
      "root": undefined,
    }
  }
  control={
    <WithStyles(Checkbox)
      checked={false}
      classes={
        Object {
          "checked": undefined,
          "root": undefined,
        }
      }
      onChange={[Function]}
    />
  }
  disabled={false}
  label={
    <WithStyles(Typography)
      classes={
        Object {
          "root": Object {
            "textAlign": "justify",
          },
        }
      }
      color="error"
    />
  }
/>
`;
